packages/app/src/containers/Canvas.tsx

```diff
@@ -711,11 +713,58 @@ export default memo(function Canvas() {
     }
   }, [state, handleDirection]);
 
+  const handleDropEvent = useCallback(
+    async (e: DragEvent) => {
+      if (!e.dataTransfer) return;
+      e.preventDefault();
+      const file = e.dataTransfer.files[0];
+      const rawPoint = getPoint(e.nativeEvent);
+      const point = offsetEventPoint(rawPoint);
+
+      if (!file) {
+        alert('Unable to read file');
+        return;
+      }
+
+      const extension = SUPPORTED_FILE_TYPES[file.type];
+
+      if (!extension) {
+        alert(
+          `Files of type ${file.type} aren't supported. Files of type png, jpg, and webp are supported.`,
+        );
+        return;
+      }
+
+      const data = await file.arrayBuffer();
+      const img = new Image();
```

Comment:
This approach is totally fine but... I think I wanna use CanvasKit for consistency after all. Plus, in the future we can share the decoded version in `ImageCache` so we only have to decode once.

```ts
const image = CanvasKit.MakeImageFromEncoded(data);

if (!image) return;

const width = image.width();
const height = image.height();

image.delete();
```

Also note that if we did go with the `createObjectURL` approach, we'd want to revoke the URL after measuring

Reply:
I forgot about that function 😅 !